src/main/bash/sdkman-install.sh

```diff
@@ -185,3 +186,34 @@ function __sdkman_validate_zip() {
 		return 1
 	fi
 }
+
+function __sdkman_checksum_zip() {
+	local -r zip_archive="$1"
+	local -r headers_archive="$2"
+	local algorithm checksum checksum_ok
+	local shasum_avail=false
+
+	#Check for the appropriate checksum tools
+	if command -v shasum > /dev/null 2>&1; then
+		shasum_avail=true
+	fi
+
+	while IFS= read -r line; do
+    	if [[ $line =~ 'x-sdkman-checksum-' ]]; then
```

Comment:
Hi @hgeraldino!

I wonder if we couldn't do part of the extraction in the if statement. Both ZSH and Bash support regex matching and groups. I'll see if I can provide you with an example later :).

I would also try to limit the amount of calls to external programs like `sed` as these are often not portable and their execution is slow (forking a subshell).

Just my 2c.

Reply:
Yeah, doing the inline pattern matching/extraction is a good idea; I'll look into that.

I picked `sed` mainly because is included on the POSIX standard set of tools, so it's expected to be available on all unix-derived OS

Reply:
Ended up removing the first if condition and just checking if the variables have been set (i.e. there was a match)

I'll look into `BASH_REMATCH` (and its zsh equivalent) for bash regex matching next